packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -245,6 +261,56 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
         startBlock_ = Storage.getUint(START_BLOCK_SLOT);
     }
 
+    /// @notice Getter for the gas paying asset address.
+    function gasPayingToken() external view returns (address addr_, uint8 decimals_) {
+        (addr_, decimals_) = GasPayingToken.getToken();
+    }
+
+    function gasPayingTokenName() external view returns (string memory name_) {
+        name_ = GasPayingToken.getName();
+    }
+
+    function gasPayingTokenSymbol() external view returns (string memory symbol_) {
+        symbol_ = GasPayingToken.getSymbol();
+    }
+
+    /// @notice Internal setter for the gas paying token address, includes validation.
+    ///         Ether uses a default value of 18 decimals and the requirement
+    ///         that the decimals are less than or equal to 18 ensures that no precision is
+    ///         lost for tokens that have less decimals.
+    /// @param _token Address of the gas paying token.
+    function _setGasPayingToken(address _token) internal {
+        if (_token == address(0)) _token = Constants.ETHER;
+
+        uint8 decimals = 18;
+        string memory name = "Ether";
+        string memory symbol = "ETH";
+        if (_token != Constants.ETHER) {
+            decimals = Token(_token).decimals();
+            require(decimals == 18, "SystemConfig: bad decimals");
+            name = Token(_token).name();
+            require(
+                abi.encodePacked(name).length <= 32,
+                "SystemConfig: name of gas paying token cannot be greater than 32 bytes"
+            );
+            symbol = Token(_token).symbol();
+            require(
+                abi.encodePacked(symbol).length <= 32,
+                "SystemConfig: symbol of gas paying token cannot be greater than 32 bytes"
+            );
+        }
+
+        if (_token != Constants.ETHER) {
+            GasPayingToken.set({ _token: _token, _decimals: decimals, _name: name, _symbol: symbol });
+            OptimismPortal(payable(optimismPortal())).setGasPayingToken({
```

Comment:
If it isn't a pain you can make the `name` and `symbol` passed all the way thru as `bytes32` since they end up being stored in a single storage slot anyways. It is a minor optimization and prevents a dos attack in the world where we did not have the above `require` statements. The spec currently defines it this way. MakerDAO used to make their tokens this way to prevent a dynamic sized returndata attack

Reply:
good idea!